Fix for choppy scrolling experience in Reader

ravishanker opened the pull request:
Fixes #14636 

Reader screen has a CollapsingToolbarLayout and a TabLayout that can feel scrolling up and down at time choppy as reported in the issue.

It is a subtle experience issue.  Seems to be due to using three flags combination on the layout_scrollFlags.  For the curious, this [post](https://medium.com/martinomburajr/android-design-collapsing-toolbar-scrollflags-e1d8a05dcb02) explains in detail.

To test:
Login to WordPress Android app.  Switch to Reader tab from the bottom navigation. Scroll up and down.  Scrolling should be is smoother than before without feeling like stuck when scrolling up or down

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Regular regression tests

3. What automated tests I added (or what prevented me from doing so)
NA

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ravishanker committed:
Fix for choppy scrolling experience in Reader

Fixes #14636

ravishanker committed:
Update reader_fragment_layout.xml

Formatting layout file, mainly for pushing for CI job

renanferrari reviewed (commented):
Thanks for taking a look at this!

After doing a little bit of testing, I'm not sure if these changes actually solve the problem.

First, regarding the use of both `enterAlways` and `enterAlwaysCollapsed` flags at the same time, I believe this is fine according to the official documentation, which [describes `enterAlwaysCollapsed` as "an additional flag" to `enterAlways`](https://developer.android.com/reference/com/google/android/material/appbar/AppBarLayout.LayoutParams#SCROLL_FLAG_ENTER_ALWAYS_COLLAPSED). 

To clarify, the expected behavior here, as far as I understand, is for the `CollapsingToolbarLayout` to show up as soon as the scrolling starts (alongside the `TabLayout`) but to be expanded only when we reach the top of the scrolling content inside the `ViewPager`. This can be demonstrated by the Notifications screen, which uses a similar layout configuration and `enterAlways|enterAlwaysCollapsed`:

![notifications-1](https://user-images.githubusercontent.com/830056/119204670-6b0a3f00-ba6c-11eb-84a8-aade2e6efd22.gif)


Second, looking at the video and description on the original issue, my understanding is that the undesirable behavior happens when trying to "fling" the content. Again, we can clearly see the difference in behavior when comparing with the Notifications screen:

|Notifications|Reader|
|---|---|
|![notifications-2](https://user-images.githubusercontent.com/830056/119204764-b45a8e80-ba6c-11eb-93a9-69c41ff15578.gif)|![following](https://user-images.githubusercontent.com/830056/119204770-bde3f680-ba6c-11eb-99d7-8d20547b3502.gif)|

Interestingly, it seems that other tabs in the Reader actually behave as expected, which makes me think this is a problem with the "Following" tab specifically. Here's an example with the "Discover" tab:

![discover](https://user-images.githubusercontent.com/830056/119204815-e3710000-ba6c-11eb-9547-927b12411bb3.gif)

I encourage you to take a second look at this, but if it looks like it's going to be too much trouble, then we should probably leave this for a HACK Week, Groundskeeping, or something like that, as this doesn't have a high priority at the moment. Let me know what you think.

ravishanker committed:
Update reader_fragment_post_cards.xml

Disable scroll gestures flag on custom FilteredRecyclerView to improve smooth scrolling on Following tab inside Reader view

ravishanker commented:
> 
> I encourage you to take a second look at this, but if it looks like it's going to be too much trouble, then we should probably leave this for a HACK Week, Groundskeeping, or something like that, as this doesn't have a high priority at the moment. Let me know what you think.

@renanferrari,  Thanks for taking deeper look into the issue.  You're right , "Discover" tab doesn't have smooth scrolling issue.  However, "Following" tab has that due to its use of a legacy custom FilteredRecylerView and Filter tab inside it that is sticky.  I've made a minor tweak by disabling the flag on FilteredRecyclerView.  However, due to the filter that is designed to be sticky at the top, it will probably will never feel like Discover tab I'm afraid.  This is the best we can achieve for now without attempting to rewrite the legacy ReaderPostListFragment.java that is ~ 3K lines!

ravishanker committed:
Update build.gradle

Testing for #14698 lint failure issue by enabling desugaring here temporarily

ravishanker committed:
Revert "Update build.gradle"

This reverts commit ec43e21eedffaac17cb21533958187dc11b8b47d.

ravishanker committed:
Fix scrolling issue

Updated with a working setting.  Reverted unnecessary

ravishanker commented:
@renanferrari I've tried the `exitUntilCollapsed` It seems to feel better and works like iOS one.  However now the top header doesn't scroll away as before (and same as iOS where it doesn't scroll away too).
I've also reverted all other changes limiting it to only this change now.  Please review again and if you're happy, merge it.  Thanks

renanferrari reviewed (approved):
Thanks for taking care of this, @ravishanker! 

LGTM 👍
